base/error.jl

```diff
@@ -50,30 +50,32 @@ macro assert(ex, msgs...)
     :($(esc(ex)) ? $(nothing) : throw(Main.Base.AssertionError($msg)))
 end
 
+const DEFAULT_RETRY_N = 1
+const DEFAULT_RETRY_ON = e->true
+const DEFAULT_RETRY_MAX_DELAY = 0
 
 """
-    retry(f, [condition]; n=3; max_delay=10) -> Function
+    retry(f, [retry_on]; n=1, max_delay=0) -> Function
 
 Returns a lambda that retries function `f` up to `n` times in the
-event of an exception. If `condition` is a `Type` then retry only
-for exceptions of that type. If `condition` is a function
-`cond(::Exception) -> Bool` then retry only if it is true.
+event of an exception. If `retry_on` is a `Type` then retry only
+for exceptions of that type. If `retry_on` is a function
+`test_error(::Exception) -> Bool` then retry only if it is true.
 
 **Examples**
 ```julia
 retry(http_get, e -> e.status == "503")(url)
 retry(read, UVError)(io)
 ```
 """
-function retry(f::Function, condition::Function=e->true;
-               n::Int=3, max_delay::Int=10)
+function retry(f::Function, retry_on::Function=DEFAULT_RETRY_ON; n=DEFAULT_RETRY_N, max_delay=DEFAULT_RETRY_MAX_DELAY)
     (args...) -> begin
         delay = 0.05
```

Comment:
See also explanations in orignal description of this feature: https://github.com/JuliaLang/julia/pull/15409#issuecomment-194018833 & https://github.com/samoconnor/julia/commit/91bde90734f340244924b1285ddd0715b8a3099c